### PR TITLE
feat(component): SfColorPicker

### LINKED ISSUE #851: [FEATURE] Implement SfColorPicker
**Describe how this feature will help us improve**
With new designs comes new components - SfColorPicker is one of them.

Designs: 
Mobile -> https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-Design-System?node-id=264%3A566

### PR DESCRIPTION
**Pull request by martaradziszewska**
# Related issue

Closes #851 

# Scope of work

- add SfColorPicker component
- add stories
- add modifier `--vertical`

# Screenshots of visual changes

# Checklist

- [ ] No commented blocks of code left
- [x] Run tests and docs
- [ ] Self code-reviewed

### CODE REVIEW

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# component-description
+ColorPicker [button](Button.html) and [color](SfColor.html)
+
+# storybook-iframe-height
+100rem
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.md:5]: 1600px? @martaradziszewska

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<template>
+  <div class="sf-color-picker" :class="{ 'sf-color-picker--active': isOpen }">
+    <transition name="sf-color-picker" mode="out-in">
+      <div v-if="!isOpen" key="1" class="sf-color-picker__open">
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:4]: @martaradziszewska IMO we should change number on key to some string value, what do you think

Comment (pspaczek) [reply packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:4]: `sf-color-picker__open` when I reed CSS I've been a little bit confused. `__open` sounds like state not like element @martaradziszewska

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        <slot name="open">
+          <SfButton
+            class="color-secondary sf-color-picker__open--button"
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:7]:
```suggestion
            class="color-secondary sf-color-picker__open-button"
```
but still we can find better name for this class 💚

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            @click="toggle"
+            >+ Colors</SfButton
+          >
+        </slot>
+      </div>
+      <div v-else key="2" class="sf-color-picker__colors">
+        <slot name="label">
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:14]: @martaradziszewska please describe all slots 👍

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  </div>
+</template>
+<script>
+import SfIcon from "../../atoms/SfIcon/SfIcon.vue";
+import SfButton from "../../atoms/SfButton/SfButton.vue";
+export default {
+  name: "SfColorPicker",
+  components: { SfIcon, SfButton },
+  props: {
+    isOpen: {
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:39]: and describe props 💚

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue (hunk 3 above)
Comment (martaradziszewska) [reply packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:7]: you're right! :) I'll change it

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          >
+        </slot>
+      </div>
+      <div v-else key="2" class="sf-color-picker__colors">
+        <slot name="label">
+          <div v-if="label" class="sf-color-picker__label">{{ label }}</div>
+        </slot>
+        <slot />
+        <slot name="close">
+          <SfIcon
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:19]: @martaradziszewska maybe we can use SfButton here?

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  props: {
+    isOpen: {
+      type: Boolean,
+      default: false,
+    },
+    label: {
+      type: String,
+      default: "",
+    },
+    closeIcon: {
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:47]: IMO we don't need it, now slot is enough, wdyt?

File: packages/vue/src/components/molecules/SfGallery/SfGallery.vue [hunk 8]
```diff
@@ -1,6 +1,7 @@
 <template>
   <div class="sf-gallery">
     <div class="sf-gallery__stage">
+      <slot name="options" />
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfGallery/SfGallery.vue:4]: @martaradziszewska I can't find the reason why you add this slot, help me, please

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue (hunk 6 above)
Comment (martaradziszewska) [reply packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:19]: 👌

File: packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue (hunk 7 above)
Comment (martaradziszewska) [reply packages/vue/src/components/molecules/SfColorPicker/SfColorPicker.vue:47]: we still need some close state whether show icon or not, right?

**Comment by pspaczek** [comment]
nice @martaradziszewska thank you 💚

**Pull request merged**